PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,880 @@
+                DEBUG)
+            self.smart_print(
+                '    OS-[shortHostName]-[protocol]-[portgroup_name]-MV',
+                DEBUG)
+            component_dict = self.get_mv_component_dict(mv_name, revert)
+            print('\n')
+        else:
+            self.smart_print('\n', DEBUG)
+            self.smart_print(
+                'Checking if masking view is in the following 2 formats: ',
+                DEBUG)
+            self.smart_print(
+                '    OS-[shortHostName]-[SRP]-[SLO]-[workload]-[protocol]-MV',
+                DEBUG)
+            self.smart_print('        OR', DEBUG)
+            self.smart_print('    OS-[shortHostName]-No_SLO-[protocol]-MV',
+                             DEBUG)
+            self.smart_print('\n', DEBUG)
+            component_dict = self.get_mv_component_dict(mv_name)
+            self.smart_print('COMPONENTS OF %s', DEBUG, mv_name)
+        if component_dict:
+            self.print_pretty_table(component_dict)
+
+        return component_dict
+
+    def check_mv_for_migration(self, mv_name, revert=False):
+        """Check if the masking view can be migrated.
+
+        :param mv_name: the masking view name
+        :param test: is it a test case
+        :returns: boolean
+        """
+        if revert:
+            component_dict = self.print_component_dict(mv_name, revert)
+            if component_dict and self.verify_protocol(
+                    component_dict['protocol']):
+                print_str = '%s HAS BEEN VERIFIED TO BEING IN THE NEW FORMAT.'
+                self.smart_print(print_str, DEBUG, mv_name)
+                return True
+            else:
+                print_str = (
+                    '%s IS NOT IN THE NEW FORMAT, MIGRATION WILL NOT '
+                    'PROCEED.')
+                self.smart_print(print_str, WARNING, mv_name)
+                return False
+        else:
+            component_dict = self.print_component_dict(mv_name)
+            if component_dict and self.verify_protocol(
+                    component_dict['protocol']):
+                print_str = '%s HAS BEEN VERIFIED TO BEING IN THE OLD FORMAT.'
+                self.smart_print(print_str, DEBUG, mv_name)
+                return True
+            else:
+                print_str = ('%s IS NOT IN THE OLD FORMAT, MIGRATION WILL NOT '
+                             'PROCEED.')
+                self.smart_print(print_str, WARNING, mv_name)
+                return False
+
+    def get_sg_component_dict(self, sg_name):
+        """Parse the storage group string.
+
+        :param sg_name: the storage group name
+        :param test: is it a test case
+        :returns: dict
+        """
+        regex_str = '^(?P<prefix>OS)-(?P<host>.+?)' \
+                    '((?P<no_slo>No_SLO)|((?P<srp>SRP.+?)-' \
+                    '(?P<sloworkload>.+?)))-(?P<portgroup>.+?)' \
+                    '(?P<after_pg>$|-CD|-RE)'
```

Comment:
`\ `used instead of `( )` for splitting across lines, should regex patterns also have `r` preceding each set of `' '` to tell the likes of pep8 it is a regex pattern?